vimiv/plugins/metadata_pyexiv2.py

```diff
@@ -0,0 +1,134 @@
+# vim: ft=python fileencoding=utf-8 sw=4 et sts=4
+
+# This file is part of vimiv.
+# Copyright 2017-2023 Christian Karl (karlch) <karlch at protonmail dot com>
+# License: GNU GPL v3, see the "LICENSE" and "AUTHORS" files for details.
+
+"""Metadata plugin based on pyexiv2 (https://pypi.org/project/py3exiv2/) backend.
+
+Properties:
+- Shared libraries as dependencies.
+- Formatted Metadata.
+- Reads Exif, IPTC and XMP.
+"""
+
+import contextlib
+import itertools
+from typing import Any, Sequence, Iterable
+
+from vimiv.imutils import metadata
+from vimiv.utils import log, is_hex, lazy
+
+pyexiv2 = lazy.import_module("pyexiv2")
+
+_logger = log.module_logger(__name__)
+
+
+class MetadataPyexiv2(metadata.MetadataPlugin):
+    """Provides metadata support based on pyexiv2."""
+
+    def __init__(self, path: str) -> None:
+        self._path = path
+
+        try:
+            self._metadata = pyexiv2.ImageMetadata(path)
+            self._metadata.read()
+        except FileNotFoundError:
+            _logger.debug("File %s not found", path)
+            self._metadata = None
+
+    @property
+    def name(self) -> str:
+        """Get the name of the used backend."""
+        return "pyexiv2"
+
+    @property
+    def version(self) -> str:
+        """Get the version of the used backend."""
+        return pyexiv2.__version__
+
+    def get_metadata(self, desired_keys: Sequence[str]) -> metadata.MetadataDictT:
+        """Get value of all desired keys for the current image."""
+
+        out = {}
+
+        if self._metadata is None:
+            return {}
+
+        for base_key in desired_keys:
+            # TODO: potentially remove
+            # For backwards compability, assume it has one of the following prefixes
```

Comment:
This goes hand in hand with the previous. I suppose it makes sense to get everything here, and finally filter in the metadata widget?

Reply:
Done (see previous conversation)